File: Labs/lab01/AquecimentoRecursivoJava/AquecimentoRecursivo.java
/* 2. Aquecimento Recursivo:
 * Refa¸ca a questão anterior de forma recursiva. Atente-se para os detalhes dos retornos de cada um dos métodos.
 */

import java.util.Scanner;

public class AquecimentoRecursivo{

    public static boolean isFIM(String str){
        return (str.length() == 3 && str.charAt(0) == 'F' && str.charAt(1) == 'I' && str.charAt(2) == 'M');
    }

    public static int contMaiusculo(String str, int indice){
        int cont = 0;
        if(indice < str.length()){
            if(str.charAt(indice) >= 'A' && str.charAt(indice) <= 'Z'){
                cont = 1 + contMaiusculo(str, ++indice);
            }
            else{
                cont = contMaiusculo(str, ++indice);
            }
        }
        return cont;
    }

    public static void main (String[] args){
        String[] strings = new String[1000];
        int numStrings = 0;
        Scanner entrada = new Scanner(System.in);

        do{
            strings[numStrings] = entrada.nextLine();
        } while ((isFIM(strings[numStrings++])) == false);
        numStrings--;

        for(int i = 0; i < numStrings; i++){
            System.out.println(contMaiusculo(strings[i], 0));
        }

        entrada.close();
    }
}

File: AAAcodigos/labs/lab01/AquecimentoRecursivoC/AquecimentoRecursivo.c
/* 2. Aquecimento Recursivo:
 * Refa¸ca a questão anterior de forma recursiva. Atente-se para os detalhes dos retornos de cada um dos métodos.
 */

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int numStrings = 0;

void str(char **str){
    char buffer[1000];

    scanf("%[^\n]", buffer);

    str[numStrings] = (char*)malloc((strlen(buffer) + 1) * sizeof(char));

    strcpy(str[numStrings], buffer);
}

int isFIM(char** str){
    if(strlen(str[numStrings]) == 3 && str[numStrings][0] == 'F' && str[numStrings][1] == 'I' && str[numStrings][2] == 'M'){
        numStrings++;
        return 0;
    }
    else{
        numStrings++;
        return 1;
    }
}

int contMaiusculo(char** str, int num, int indice){
    int cont = 0;

    if(indice < strlen(str[num])){
            if(str[num][indice] >= 'A' && str[num][indice] <= 'Z'){
                cont = 1 + contMaiusculo(str, num, ++indice);
            }
            else{
                cont = contMaiusculo(str, num, ++indice);
            }
    }
    return cont;
}
        

int main(){
    char **strings;

    strings = (char**)malloc(1000 * sizeof(char*));

    do{
        str(strings);
        getchar();
    } while (isFIM(strings) == 1);
    numStrings--;

    for(int i = 0; i < numStrings; i++){
        printf("%d\n", contMaiusculo(strings, i, 0));
    }
}